### PR TITLE
Freddy/Add module owner with tx and query

### PR DESCRIPTION
**Pull request by freddyli7**
## Description

- add `add-genesis-module-owner` CLI tx to add init module owner
- add module genesis and write init module owner into store after chain launched

### CODE REVIEW

File: proto/chainlink/v1beta/genesis.proto [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
-// this line is used by starport scaffolding # genesis/proto/import
+import "gogoproto/gogo.proto";
 
 option go_package = "github.com/ChainSafe/chainlink-cosmos/x/chainlink/types";
 
-// GenesisState defines the chainlink module's genesis state.
 message GenesisState {
-    // this line is used by starport scaffolding # genesis/proto/state
-    // this line is used by starport scaffolding # ibc/genesis/proto
+  // moduleOwners is an array containing the chainlink init module owner accounts.
```
Comment (araskachoi) [inline proto/chainlink/v1beta/genesis.proto:9]: nice!

File: x/chainlink/client/cli/txModuleInfo.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import (
+	"errors"
+	"github.com/ChainSafe/chainlink-cosmos/x/chainlink/types"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/spf13/cobra"
+)
+
+func CmdAddModule() *cobra.Command {
```
Comment (araskachoi) [inline x/chainlink/client/cli/txModuleInfo.go:12]: lets rename to `CmdAddModuleOwner()`

File: x/chainlink/handler.go [hunk 3]
```diff
@@ -16,9 +16,38 @@ func NewHandler(k keeper.Keeper) sdk.Handler {
...
+	if err != nil {
+		return nil, err
+	}
+	result, err := sdk.WrapServiceResult(ctx, msgResult, err)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
```
Comment (araskachoi) [inline x/chainlink/handler.go:53]: gj moving these from `x/chainlink/handlerMsgSubmitFeed.go`! makes more sense here

File: x/chainlink/keeper/keeper.go [hunk 4]
```diff
@@ -16,24 +16,27 @@ import (
...
 type (
 	Keeper struct {
-		cdc           codec.Marshaler
-		feedStoreKey  sdk.StoreKey
-		roundStoreKey sdk.StoreKey
-		memKey        sdk.StoreKey
+		cdc            codec.Marshaler
+		feedStoreKey   sdk.StoreKey
+		roundStoreKey  sdk.StoreKey
+		moduleStoreKey sdk.StoreKey
```
Comment (araskachoi) [inline x/chainlink/keeper/keeper.go:22]: let's rename to `moduleOwnerStoreKey`

File: x/chainlink/keeper/keeper.go [hunk 5]
```diff
@@ -16,24 +16,27 @@ import (
...
+		memKey         sdk.StoreKey
 	}
 )
 
 func NewKeeper(
 	cdc codec.Marshaler,
 	feedStoreKey,
 	roundStoreKey,
+	moduleStoreKey,
```
Comment (araskachoi) [inline x/chainlink/keeper/keeper.go:31]: `moduleOwnerStoreKey`

File: x/chainlink/keeper/keeper.go [hunk 6]
```diff
@@ -16,24 +16,27 @@ import (
...
 ) *Keeper {
 	return &Keeper{
-		cdc:           cdc,
-		feedStoreKey:  feedStoreKey,
-		roundStoreKey: roundStoreKey,
-		memKey:        memKey,
+		cdc:            cdc,
+		feedStoreKey:   feedStoreKey,
+		roundStoreKey:  roundStoreKey,
+		moduleStoreKey: moduleStoreKey,
```
Comment (araskachoi) [inline x/chainlink/keeper/keeper.go:38]: `moduleOwnerStoreKey`

File: x/chainlink/genesis.go [hunk 7]
```diff
@@ -6,21 +6,22 @@ import (
...
 func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState) {
-	// this line is used by starport scaffolding # genesis/module/init
-
-	// this line is used by starport scaffolding # ibc/genesis/init
+	for _, owner := range genState.GetModuleOwners() {
+		m := types.ModuleOwner{
+			Address: owner.GetAddress(),
+			PubKey:  owner.GetPubKey(),
+		}
+		k.SetModuleOwner(ctx, &m)
```
Comment (araskachoi) [inline x/chainlink/genesis.go:16]: amazing. first instance of the add module owner for genesis account. great!!

File: x/chainlink/types/msg.go [hunk 8]
```diff
@@ -56,3 +57,47 @@ func (m *MsgFeedData) ValidateBasic() error {
...
+	// TODO: add proper cosmos address and pubkey validation
+	return nil
+}
+
+func (m *ModuleOwner) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(m)
+	return sdk.MustSortJSON(bz)
+}
+
+func (m *ModuleOwner) GetSigners() []sdk.AccAddress {
```
Comment (araskachoi) [inline x/chainlink/types/msg.go:88]: would there ever be the case where there will be more than one `m.Address` set in a `ModuleOwner` obj?

**Review by araskachoi — APPROVED** [review]

File: x/chainlink/client/cli/txModuleInfo.go (hunk 2 above)
Comment (freddyli7) [reply x/chainlink/client/cli/txModuleInfo.go:12]: yes

File: x/chainlink/keeper/keeper.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (freddyli7) [reply x/chainlink/keeper/keeper.go:22]: yes makes sense

**Pull request merged**